app/views/tracks/show/joined.html.haml

```diff
@@ -15,6 +15,35 @@
   %article
     .lg-container.container
       .lhs
+        %section.activities
+          - @activities.each do |activity|
+            = link_to activity.url, class: 'activity' do
+              - if activity.is_a?(User::Activities::StartedExerciseActivity)
+                = graphical_icon :"check-circle", css_class: "descriptive-icon"
+                .info
+                  .description
+                    = activity.text
+                    .exercise
+                      = exercise_icon(activity.exercise)
+                    .exercise_name= activity.exercise.title
+                  %time{ datetime: activity.occurred_at } #{time_ago_in_words(activity.occurred_at)} ago
+                = graphical_icon :"chevron-right", css_class: "action-icon"
+
+              - elsif activity.is_a?(User::Activities::SubmittedIterationActivity)
+                = graphical_icon :"check-circle", css_class: "descriptive-icon"
+                .info
+                  .description
+                    = activity.text
+                    .exercise
+                      = exercise_icon(activity.exercise)
+                    .exercise_name= activity.exercise.title
+
+                  .iteration
+                    = ViewComponents::Track::IterationSummary.new(activity.iteration)
+                    = graphical_icon :"chevron-right", css_class: "action-icon"
+
+                  %time{ datetime: activity.occurred_at } #{time_ago_in_words(activity.occurred_at)} ago
+
```

Comment:
This will all get split out into view components, but I've not yet decided whether they should be haml-side or React, so for now, I've just inlined.